calcs/source.py

```diff
@@ -5,105 +5,249 @@
+        elif circular is False:
+            circular_mask = self.ecc > self.ecc_tol
+        else:
+            raise ValueError("`circular` must be None, True or False")
+
+        if stationary is None:
+            stationary_mask = np.repeat(True, self.n_sources)
+        elif stationary is True:
+            stationary_mask = utils.determine_stationarity(self.m_1, self.m_2,
+                                                           self.f_orb, t_obs,
+                                                           self.ecc, self.stat_tol)
+        elif stationary is False:
+            stationary_mask = np.logical_not(utils.determine_stationarity(self.m_1, self.m_2,
+                                                           self.f_orb, t_obs,
+                                                           self.ecc, self.stat_tol))
+        else:
+            raise ValueError("`stationary` must be None, True or False")
+
+        return np.logical_and(circular_mask, stationary_mask)
+
+    def get_snr(self, t_obs=4 * u.yr, max_harmonic=50, n_step=100):
+        """Computes the SNR for a generic binary
 
         Params
         ------
         t_obs : `array`
-            observation duration in units of yr
+            observation duration (default: 4 years)
 
         ecc_tol : `float`
             tolerance for treating a binary as eccentric
 
-        n_max : `int`
+        stat_tol : `float`
+            tolerance for treating a binary as stationary
+            (using fractional change in frequency)
+
+        max_harmonic : `int`
             maximum integer harmonic to consider for eccentric sources
 
+        n_step : `int`
+            number of time steps during observation duration
+
         Returns
         -------
         SNR : `array`
             the signal to noise ratio
         """
-        t_obs = t_obs * u.yr
-        m_c = utils.chirp_mass(m_1=self.m_1, m_2=self.m_2)
-        snr = np.zeros(len(self.m_1))
-        ind_ecc, = np.where(self.ecc > ecc_tol)
-        ind_circ, = np.where(self.ecc <= ecc_tol)
-
-        # Treat circular first
-        snr[ind_circ] = sn.snr_circ_stationary(m_c=m_c[ind_circ].to(u.kg), 
-                                               f_orb=self.f_orb[ind_circ], 
-                                               dist=self.dist[ind_circ].to(u.m), 
-                                               t_obs=t_obs.to(u.s))
-
-        snr[ind_ecc] = sn.snr_ecc_stationary(m_c=m_c[ind_ecc].to(u.kg),
-                                             f_orb=self.f_orb[ind_ecc],
-                                             ecc=self.ecc[ind_ecc],
-                                             dist=self.dist[ind_ecc].to(u.m),
-                                             t_obs=t_obs.to(u.s),
-                                             n_max=n_max)
-
+        
+        snr = np.zeros(self.n_sources)
+        stationary_mask = self.get_source_mask(circular=None, stationary=True, t_obs=t_obs)
```

Comment:
this is so much more elegant!

Reply:
Thanks! This makes me realise I should be using this function for the circular masks later as well